From dc41591d5ceb18900ec85894f8f7b7bb44bb3bd9 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Mon, 4 Jan 2016 01:44:04 -0500
Subject: [PATCH] avoid searching host dirs

Don't search the hardcoded host dirs to avoid
host contamination.

Upstream-Status: Inappropriate [cross-compile specific]

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 acinclude.m4                                                | 4 ++--
 src/modules/rlm_sql/drivers/rlm_sql_db2/configure.ac        | 4 ++--
 src/modules/rlm_sql/drivers/rlm_sql_firebird/configure.ac   | 4 ++--
 src/modules/rlm_sql/drivers/rlm_sql_iodbc/configure.ac      | 4 ++--
 src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.ac      | 6 +++---
 src/modules/rlm_sql/drivers/rlm_sql_oracle/configure.ac     | 2 +-
 src/modules/rlm_sql/drivers/rlm_sql_postgresql/configure.ac | 4 ++--
 src/modules/rlm_sql/drivers/rlm_sql_unixodbc/configure.ac   | 4 ++--
 8 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index da48acc..b513ae1 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -178,7 +178,7 @@ if test "x$smart_lib" = "x"; then
   FR_LOCATE_DIR(smart_lib_dir,[lib$1${libltdl_cv_shlibext}])
   FR_LOCATE_DIR(smart_lib_dir,[lib$1.a])
 
-  for try in $smart_lib_dir /usr/local/lib /opt/lib; do
+  for try in $smart_lib_dir; do
     AC_MSG_CHECKING([for $2 in -l$1 in $try])
     LIBS="-l$1 $old_LIBS"
     CPPFLAGS="-L$try -Wl,-rpath,$try $old_CPPFLAGS"
@@ -218,7 +218,7 @@ ac_safe=`echo "$1" | sed 'y%./+-%__pm%'`
 old_CPPFLAGS="$CPPFLAGS"
 smart_include=
 dnl #  The default directories we search in (in addition to the compilers search path)
-smart_include_dir="/usr/local/include /opt/include"
+smart_include_dir=
 
 dnl #  Our local versions
 _smart_try_dir=
diff --git a/src/modules/rlm_sql/drivers/rlm_sql_db2/configure.ac b/src/modules/rlm_sql/drivers/rlm_sql_db2/configure.ac
index 75c851a..a262d71 100644
--- a/src/modules/rlm_sql/drivers/rlm_sql_db2/configure.ac
+++ b/src/modules/rlm_sql/drivers/rlm_sql_db2/configure.ac
@@ -57,14 +57,14 @@ if test x$with_[]modname != xno; then
 		esac])
 
 	dnl Check for SQLConnect in -ldb2
-	smart_try_dir="$ibmdb2_lib_dir /usr/local/db2/lib /usr/IBMdb2/V7.1/lib"
+	smart_try_dir="$ibmdb2_lib_dir"
 	FR_SMART_CHECK_LIB(db2, SQLConnect)
 	if test "x$ac_cv_lib_db2_SQLConnect" != xyes; then
 		fail="$fail libdb2"
 	fi
 
 	dnl Check for sqlcli.h
-	smart_try_dir="$ibmdb2_include_dir /usr/local/db2/include /usr/IBMdb2/V7.1/include"
+	smart_try_dir="$ibmdb2_include_dir"
 	FR_SMART_CHECK_INCLUDE(sqlcli.h)
 	if test "x$ac_cv_header_sqlcli_h" != xyes; then
 		fail="$fail sqlcli.h"
diff --git a/src/modules/rlm_sql/drivers/rlm_sql_firebird/configure.ac b/src/modules/rlm_sql/drivers/rlm_sql_firebird/configure.ac
index 4da57b3..752b043 100644
--- a/src/modules/rlm_sql/drivers/rlm_sql_firebird/configure.ac
+++ b/src/modules/rlm_sql/drivers/rlm_sql_firebird/configure.ac
@@ -56,14 +56,14 @@ if test x$with_[]modname != xno; then
 		esac])
 
 	dnl Check for isc_attach_database in -lfbclient
-	smart_try_dir="$firebird_lib_dir /usr/lib/firebird2/lib /usr/local/firebird/lib"
+	smart_try_dir="$firebird_lib_dir"
 	FR_SMART_CHECK_LIB(fbclient, isc_attach_database)
 	if test "x$ac_cv_lib_fbclient_isc_attach_database" != xyes; then
 		fail="$fail libfbclient"
 	fi
 
 	dnl Check for ibase.h
-	smart_try_dir="$firebird_include_dir /usr/lib/firebird2/include /usr/local/firebird/include"
+	smart_try_dir="$firebird_include_dir"
 	FR_SMART_CHECK_INCLUDE(ibase.h)
 	if test "x$ac_cv_header_ibase_h" != xyes; then
 		fail="$fail ibase.h"
diff --git a/src/modules/rlm_sql/drivers/rlm_sql_iodbc/configure.ac b/src/modules/rlm_sql/drivers/rlm_sql_iodbc/configure.ac
index ba6304f..3393557 100644
--- a/src/modules/rlm_sql/drivers/rlm_sql_iodbc/configure.ac
+++ b/src/modules/rlm_sql/drivers/rlm_sql_iodbc/configure.ac
@@ -57,14 +57,14 @@ if test x$with_[]modname != xno; then
 		esac])
 
 	dnl Check for SQLConnect in -liodbc
-	smart_try_dir="$iodbc_lib_dir /usr/lib /usr/lib/iodbc /usr/local/lib/iodbc /usr/local/iodbc/lib/iodbc"
+	smart_try_dir="$iodbc_lib_dir"
 	FR_SMART_CHECK_LIB(iodbc, SQLConnect)
 	if test "x$ac_cv_lib_iodbc_SQLConnect" != xyes; then
 		fail="$fail libiodbc"
 	fi
 
 	dnl Check for isql.h
-	smart_try_dir="$iodbc_include_dir /usr/include /usr/include/iodbc /usr/local/iodbc/include"
+	smart_try_dir="$iodbc_include_dir"
 	FR_SMART_CHECK_INCLUDE(isql.h)
 	if test "x$ac_cv_header_isql_h" != xyes; then
 		fail="$fail isql.h"
diff --git a/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.ac b/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.ac
index 1401677..2e7db44 100644
--- a/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.ac
+++ b/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.ac
@@ -136,7 +136,7 @@ if test x$with_[]modname != xno; then
 
 	dnl # Check for libmysqlclient_r
 	if test "x$have_a_libmysqlclient" != "xyes"; then
-	    smart_try_dir="$mysql_lib_dir /usr/lib /usr/lib/mysql /usr/local/lib/mysql /usr/local/mysql/lib/mysql"
+	    smart_try_dir="$mysql_lib_dir"
 	    FR_SMART_CHECK_LIB(mysqlclient_r, mysql_init)
 	    if test "x$ac_cv_lib_mysqlclient_r_mysql_init" = "xyes"; then
 			have_a_libmysqlclient='yes'
@@ -145,7 +145,7 @@ if test x$with_[]modname != xno; then
 
 	dnl # Check for libmysqlclient
 	if test "x$have_a_libmysqlclient" != "xyes"; then
-	    smart_try_dir="$mysql_lib_dir /usr/lib /usr/lib/mysql /usr/local/lib/mysql /usr/local/mysql/lib/mysql"
+	    smart_try_dir="$mysql_lib_dir"
 	    FR_SMART_CHECK_LIB(mysqlclient, mysql_init)
 	    if test "x$ac_cv_lib_mysqlclient_mysql_init" = "xyes"; then
 			have_a_libmysqlclient='yes'
@@ -189,7 +189,7 @@ if test x$with_[]modname != xno; then
     fi
 
     if test "x$have_mysql_h" != "xyes"; then
-		smart_try_dir="$mysql_include_dir /usr/local/include /usr/local/mysql/include"
+		smart_try_dir="$mysql_include_dir"
 		FR_SMART_CHECK_INCLUDE(mysql/mysql.h)
 		if test "x$ac_cv_header_mysql_mysql_h" = "xyes"; then
 	    	AC_DEFINE(HAVE_MYSQL_MYSQL_H, [], [Define if you have <mysql/mysql.h>])
diff --git a/src/modules/rlm_sql/drivers/rlm_sql_oracle/configure.ac b/src/modules/rlm_sql/drivers/rlm_sql_oracle/configure.ac
index 3178462..5cbc8c2 100644
--- a/src/modules/rlm_sql/drivers/rlm_sql_oracle/configure.ac
+++ b/src/modules/rlm_sql/drivers/rlm_sql_oracle/configure.ac
@@ -63,7 +63,7 @@ if test x$with_[]modname != xno; then
     dnl # Check for header files
     dnl ############################################################
 
-    smart_try_dir="$oracle_include_dir /usr/local/instaclient/include"
+    smart_try_dir="$oracle_include_dir"
 
     if test "x$ORACLE_HOME" != "x"; then
 	smart_try_dir="${smart_try_dir} ${ORACLE_HOME}/include"
diff --git a/src/modules/rlm_sql/drivers/rlm_sql_postgresql/configure.ac b/src/modules/rlm_sql/drivers/rlm_sql_postgresql/configure.ac
index 4f9a890..e1cf811 100644
--- a/src/modules/rlm_sql/drivers/rlm_sql_postgresql/configure.ac
+++ b/src/modules/rlm_sql/drivers/rlm_sql_postgresql/configure.ac
@@ -41,7 +41,7 @@ if test x$with_[]modname != xno; then
 	  esac ]
 	)
 
-	smart_try_dir="$rlm_sql_postgresql_include_dir /usr/include/postgresql /usr/local/pgsql/include /usr/include/pgsql"
+	smart_try_dir="$rlm_sql_postgresql_include_dir"
 	FR_SMART_CHECK_INCLUDE(libpq-fe.h)
 	if test "x$ac_cv_header_libpqmfe_h" != "xyes"; then
 		fail="$fail libpq-fe.h"
@@ -76,7 +76,7 @@ if test x$with_[]modname != xno; then
 		  ])
 	fi
 
-	smart_try_dir="$rlm_sql_postgresql_lib_dir /usr/lib /usr/local/pgsql/lib"
+	smart_try_dir="$rlm_sql_postgresql_lib_dir"
 	FR_SMART_CHECK_LIB(pq, PQconnectdb)
 	if test "x$ac_cv_lib_pq_PQconnectdb" != "xyes"; then
 		fail="$fail libpq"
diff --git a/src/modules/rlm_sql/drivers/rlm_sql_unixodbc/configure.ac b/src/modules/rlm_sql/drivers/rlm_sql_unixodbc/configure.ac
index 3545387..c543ed4 100644
--- a/src/modules/rlm_sql/drivers/rlm_sql_unixodbc/configure.ac
+++ b/src/modules/rlm_sql/drivers/rlm_sql_unixodbc/configure.ac
@@ -57,14 +57,14 @@ if test x$with_[]modname != xno; then
 		esac])
 
 	dnl Check for SQLConnect in -lodbc
-	smart_try_dir="$unixodbc_lib_dir /usr/local/unixodbc/lib"
+	smart_try_dir="$unixodbc_lib_dir"
 	FR_SMART_CHECK_LIB(odbc, SQLConnect)
 	if test "x$ac_cv_lib_odbc_SQLConnect" != xyes; then
 		fail="$fail libodbc"
 	fi
 
 	dnl Check for sql.h
-	smart_try_dir="$unixodbc_include_dir /usr/local/unixodbc/include"
+	smart_try_dir="$unixodbc_include_dir"
 	FR_SMART_CHECK_INCLUDE(sql.h)
 	if test "x$ac_cv_header_sql_h" != xyes; then
 		fail="$fail sql.h"
-- 
1.9.1

